docs/decisions/0001-avoid-primative-constraint-types.md

```diff
@@ -0,0 +1,96 @@
+---
+status: accepted
+date: 2024-11-29
+decision-makers: eljobe@ plasmapower@
+---
+
+# Avoid primative constraint types
```

Comment:
s/primative/primitive/ here and the filename

Reply:
Done.